agent-inject/agent/annotations.go

```diff
@@ -589,60 +589,79 @@ func Init(pod *corev1.Pod, cfg AgentConfig) error {
 func (a *Agent) secrets() []*Secret {
 	var secrets []*Secret
 
-	for name, path := range a.Annotations {
-		secretName := fmt.Sprintf("%s-", AnnotationAgentInjectSecret)
-		if strings.Contains(name, secretName) {
-			raw := strings.ReplaceAll(name, secretName, "")
-			name := raw
-
-			if ok, _ := a.preserveSecretCase(raw); !ok {
-				name = strings.ToLower(raw)
-			}
-
-			if name == "" {
+	for annotationName, annotationValue := range a.Annotations {
```

Comment:
I think if someone specifies multiple annotations from the `switch` e.g. _both_ `vault.hashicorp.com/agent-inject-secret-template-file-<name>` and `vault.hashicorp.com/agent-inject-secret-<name>`, we'll add two entries to the `secrets` slice. I'm not sure how that will shake out further along in the execution path, but it would be good to add a test and guard against that to avoid unexpected behaviour.

Reply:
So I thought about that, and the only one solution I found is to prioritize file template and template annotation over secret annotation, so injector will skip `template` if `file-template` specified, and also skip `secret` if `template` specified

Reply:
@tomhjp what do you think?